GitCommands/Settings/AppSettings.cs

```diff
@@ -1626,6 +1629,29 @@ public static void LoadSettings()
             }
         }
 
+        public static void MigrateBuildServerIntegrationSettings()
+        {
+            try
+            {
+                const string settingsName = "__MigrateBuildServerIntegrationSettings";
+
+                bool migrated = GetBool(settingsName, false);
+
+                if (!migrated)
+                {
+                    string settingsFileText = File.ReadAllText(SettingsFilePath);
+                    string result = settingsFileText.Replace("BuildServer.", $"{BuildServerIntegrationPluginId}.");
+                    File.WriteAllText(SettingsFilePath, result);
+                }
+
+                SetBool(settingsName, true);
+            }
+            catch
+            {
+                // ignored
+            }
+        }
```

Comment:
This is a very blunt approach, and it will break users who run different versions side-by-side, e.g. v3.x and v4.x. Besides in which settings file are we replacing this? We have several layers of configs - the global (`C:\Users\<user>\AppData\Roaming\GitExtensions\GitExtensions\GitExtensions.settings`), the team (`<repo root>\GitExtensions.settings`) and the user (`<repo root>\.git\GitExtensions.settings`).

Whenever we open a repo we need to do something like this:
- load the config, 
- check if settings were migrated
- if not:
    -  read the original settings
    - write the new settings (to the settings at the same layer)
    - write the flag indicating that the settings were migrated (to the settings at the same layer)
- if yes - read the new settings